packages/syft/src/syft/core/adp/data_subject_ledger.py

```diff
@@ -115,43 +115,40 @@ def get_unique_data_subjects(data_subjects_query: np.ndarray) -> np.ndarray:
 
 def convert_dsa_to_index_array(
     data_subject_array: np.ndarray,
-) -> Tuple[np.ndarray, int]:
+    rdp_constants: Dict[str, np.ndarray],
+    constant: np.ndarray,
+) -> Dict[str, np.ndarray]:
     """Convert data subject array to data subject index array."""
 
     unique_data_subjects = get_unique_data_subjects(data_subject_array)
-    max_entity = len(unique_data_subjects)
 
-    input_entities_indexes_list: List[np.ndarray] = []
-
-    for data_subject_idx, data_subject in enumerate(unique_data_subjects):
+    for data_subject_name in unique_data_subjects:
         # Create a mask where the current data subject is present
-        data_subject = DataSubjectArray([data_subject])
+        data_subject = DataSubjectArray([data_subject_name])
         ds_mask = np.isin(data_subject_array, data_subject)
```

Comment:
I dont remember if @shubham3121 merged his PR but I recall there was a problem with this np.isin for nested data_subject in groups

Reply:
@teo-milea that change is merged in your fix publish loop branch. So it should be fine once that PR is merged.

Reply:
great! sorry that I forgot :laughing: